x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -280,6 +291,9 @@ export class Embeddable
     if (!this.savedVis || !this.isInitialized) {
       return;
     }
+    if (this.input.onLoad) {
```

Comment:
This can cause the loading state to become stuck - if the expression (or search context) doesn't change, the chart won't actually re-render. Looking through the code again, I think the safest way is to introduce a new callback to expose the `isLoading` state of the expression renderer component (https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/expressions/public/react_expression_renderer.tsx#L45)